File: amazia-core/src/main/java/com/example/notice/dto/UpdateNoticeRequest.java
package com.example.notice.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDateTime;

/**
 * Console からのお知らせ編集リクエスト（フェーズ19 r2）。
 *
 * <p>項目は CreateNoticeRequest と同じだが、編集での意図を明示するため別クラスとして分離する。
 * 値の解釈・補完は Console 側で完了している前提。
 */
public record UpdateNoticeRequest(
        @NotBlank
        @Size(min = 1, max = 255, message = "subject: 1〜255 文字")
        String subject,

        @NotNull
        Long categoryId,

        @NotBlank
        @Size(min = 1, max = 10000, message = "body: 1〜10000 文字")
        String body,

        @NotNull
        LocalDateTime publishStart,

        @NotNull
        LocalDateTime publishEnd
) {
}

File: amazia-console/app/Notice/Request/UpdateNoticeRequest.php
<?php

namespace App\Notice\Request;

use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Validation\Rule;

/**
 * フェーズ19: お知らせ編集 FormRequest（規約 4-1 / config 駆動）。
 */
class UpdateNoticeRequest extends FormRequest
{
    public function authorize(): bool
    {
        return true;
    }

    public function rules(): array
    {
        return [
            'subject'      => ['required', 'string', 'min:1', 'max:'.config('app.notice.subject_max_length')],
            'categoryId'   => ['required', 'integer', Rule::in([
                config('app.notice.categories.important_id'),
                config('app.notice.categories.normal_id'),
            ])],
            'body'         => ['required', 'string', 'min:1', 'max:'.config('app.notice.body_max_length')],
            'publishStart' => ['required', 'date'],
            'publishEnd'   => ['required', 'date', 'after_or_equal:publishStart'],
        ];
    }
}
